src/app/services/articles.service.ts

```diff
@@ -0,0 +1,53 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, map, distinctUntilChanged } from "rxjs";
+import { Article } from "../models/article.model";
+import { ARTICLES } from '../models/articles.model';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ArticlesService {
+  private articles = new BehaviorSubject(ARTICLES);
+
+  articles$ = this.articles.asObservable();
+
+  popularTags$ = this.articles$.pipe(
+    distinctUntilChanged((previous, next) => previous.length === next.length),
```

Comment:
em, it could produce an error if list changed, but length is the same(pop and push)